core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -224,15 +242,28 @@ public DirectoryWatcher(
    * is closed.
    */
   public void watch() {
+    final AtomicInteger eventCount = new AtomicInteger(0);
     while (listener.isWatching()) {
       // wait for key to be signalled
       WatchKey key;
       try {
-        key = watchService.take();
+        key = watchService.poll();
+        if (key == null) {
+          if (timeout > 0) {
+            currentOnIdleTimerTask = new OnIdleTimerTask(listener, eventCount.get());
```

Comment:
What if this instance already exists from the previous loop iteration? Maybe it should be cancelled and nulled first?

Reply:
fixed